files/setup-011-fluentbit.sh

```diff
@@ -0,0 +1,246 @@
+#!/bin/bash
+# Copyright 2019 VMware, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-2
+
+# Setup FluentBit
+
+set -euo pipefail
+
+# Retrieve the FluentBit image
+VEBA_BOM_FILE=/root/config/veba-bom.json
+FLUENTBIT_IMAGE=$(jq -r < ${VEBA_BOM_FILE} '.["fluentbit"].containers[0].name')
+FLUENTBIT_VERSION=$(jq -r < ${VEBA_BOM_FILE} '.["fluentbit"].containers[0].version')
+
+cat > /root/config/fluentbit-preperations.yaml << __FLUENTBIT_PREPERATIONS__
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: fluent-bit
+  namespace: vmware-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: fluent-bit-read
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - namespaces
+  - pods
+  verbs:
+  - get
+  - list
+  - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: fluent-bit-read
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: fluent-bit-read
+subjects:
+- kind: ServiceAccount
+  name: fluent-bit
+  namespace: vmware-system
+__FLUENTBIT_PREPERATIONS__
+
+cat > /root/config/fluentbit-configmap.yaml << __FLUENTBIT_CONFIGMAP__
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: fluent-bit-config
+  namespace: vmware-system
+  labels:
+    k8s-app: fluent-bit
```

Comment:
you might want to add these labels to all related objects, incl. RBAC

Reply:
Good catch. I will adjust it.